src/main/java/scmspain/karyon/restrouter/core/ParamTypeResolver.java

```diff
@@ -16,11 +16,11 @@ public T resolveValueType(Class<T> classType, String value){
     } else if (classType.isAssignableFrom(boolean.class) || classType.isAssignableFrom(Boolean.class)) {
       return (T)Boolean.valueOf(value);
     } else if (classType.isAssignableFrom(long.class) || classType.isAssignableFrom(Long.class)) {
-      return (T)Boolean.valueOf(value);
+      return (T)Long.valueOf(value);
     } else if (classType.isAssignableFrom(short.class) || classType.isAssignableFrom(Short.class)) {
-      return (T)Boolean.valueOf(value);
+      return (T)Short.valueOf(value);
     } else if (classType.isAssignableFrom(float.class) || classType.isAssignableFrom(Float.class)) {
-      return (T)Boolean.valueOf(value);
+      return (T)Float.valueOf(value);
     }
     return null;
```

Comment:
Thats another different thing...we would need to provide a hook for the user make their own construction of the object and not being coupled to library for deserialization or whatever...